xarray/backends/h5netcdf_.py

```diff
@@ -123,10 +130,6 @@ def prepare_variable(self, name, variable, check_encoding=False,
         if dtype is str:
             dtype = h5py.special_dtype(vlen=unicode_type)
 
-        if unlimited_dims is not None:
-            warnings.warn('h5netcdf does not support unlimited dimensions, '
-                          'got: %s.' % unlimited_dims)
-            unlimited_dims = None
         self.set_necessary_dimensions(variable, unlimited_dims=unlimited_dims)
```

Comment:
You will need to set dimensions to their current size explicitly with `resize_dimension()`. h5netcdf doesn't support automatic enlargement with assignment yet.

Reply:
Yes, that makes sense. I was wondering where that belonged.